### PR TITLE
Add option to ignore accessors in adjacent-overload-signatures

### PR DESCRIPTION
**Pull request by saberduck**
#### PR checklist

- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:
This change adds option to ignore getters / setters when considering overloads in adjacent-overload-signatures to allow grouping getters and setters together.

#### CHANGELOG.md entry:

[new-rule-option] `"ignore-accessors"` for [`adjacent-overload-signatures`](https://palantir.github.io/tslint/rules/adjacent-overload-signatures/) to ignore getters / setters overloads

### CODE REVIEW

File: src/rules/adjacentOverloadSignaturesRule.ts [hunk 1]
```diff
@@ -20,14 +20,25 @@ import * as ts from "typescript";
...
         ruleName: "adjacent-overload-signatures",
         description: "Enforces function overloads to be consecutive.",
-        optionsDescription: "Not configurable.",
-        options: null,
-        optionExamples: [true],
+        optionsDescription: Lint.Utils.dedent`
+            If \`${IGNORE_ACCESSORS}\` is specified, then getters and setters are not considered to be overloads
+            of function with same signature.`,
+        options: {
+            type: "string",
```
Comment (giladgray) [inline src/rules/adjacentOverloadSignaturesRule.ts]: this suggests that the option is required (but only accepts one value). can you please confirm the usage here?

Comment (saberduck) [reply src/rules/adjacentOverloadSignaturesRule.ts]: @giladgray in fact, this probably should be an array, like in [memberAccessRule.ts#L63](https://github.com/palantir/tslint/blob/master/src/rules/memberAccessRule.ts#L63)

**Comment by saberduck** [comment]
Because getters/setters are not overloads, I am thinking if this behavior should be configurable at all. Maybe the rule should be just changed to ignore getters/setters?  If you agree I can remove this PR and create a new one

**Comment by giladgray** [comment]
@saberduck ah great point about them not being overloads. i actually like how the rule works currently, where it keeps `get x()` and `set x()` together. i suppose an option to disable that could be nice though.

**Comment by saberduck** [comment]
@giladgray can you please review again?

**Comment by saberduck** [comment]
It seems that new precommit hook reformats `tslint.json` file and it differs from master, how can I avoid that?

**Comment by JoshuaKGoldberg** [comment]
@saberduck yeah that problem's on TSLint's end; it's not a problem if your PR contains the changes. Prettier was enabled before everything was merged, so there are still a few files that need to be manually reformatted...

**Comment by JoshuaKGoldberg** [comment]
Sorry for the delay @saberduck! I'm applying the formatting & merge changes now, and will merge this in once the build passes. 

Just a heads up - I'm also changing the rule's options to be of the form `{ "ignore-accessors": true }`. Other rules have had pains trying to mix and match string configurations with more complex option objects. It's been easier to stick with options objects like this.

**Pull request merged**